etna/transforms/timestamp/holiday.py

```diff
@@ -84,13 +138,24 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         :
             pd.DataFrame with added holidays
         """
-        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
-            raise ValueError("Frequency of data should be no more than daily.")
+        numbers_part = "".join(itertools.takewhile(str.isdigit, self.freq))
+        letters_part = "".join(itertools.dropwhile(str.isdigit, self.freq))
+        numbers_part = "1" if numbers_part == "" else numbers_part
+        if bigger_than_day(self.freq) and self._mode is not HolidayTransformMode.days_count:
+            raise ValueError("In default mode frequency of data should be no more than daily.")
 
         cols = df.columns.get_level_values("segment").unique()
         out_column = self._get_column_name()
 
-        if self._mode is HolidayTransformMode.category:
+        if self._mode is HolidayTransformMode.days_count:
+            encoded_matrix = np.empty(0)
+            for date in df.index:
```

Comment:
I think we should replace variable name "date" with something else (e.g. dt), because we can meet import like this:
```python
from datetime import date
```